src/config-loader/env.ts

```diff
@@ -23,6 +23,7 @@ export const cache = {
 export const db = {
 	poolSize: 50,
 	idleTimeoutMillis: 30000,
+	connectionTimeoutMillis: 30000,
```

Comment:
What is this timeout for exactly? Is there any way for it to unexpectedly close connections, eg. in the middle of a transaction?

Reply:
No, this is for initially creating a connection, eg `pool.connect()` - right now it will wait forever so if we use up all of the pool then things will start to just appear to hang, this will just cause connection attempts that have been waiting for too long to fail (which they almost certainly would anyway due to request timeouts/etc)

Reply:
I see, that makes sense, thanks. Is a much shorter timeout, maybe around 5 seconds, reasonable? I mean if it takes 5 seconds to get a free connection from the pool you're already in a pretty bad state, having the timeout at 30 feels like things could really get clogged up.

Reply:
And, incidentally, 30 seconds is already much more than most API clients have for their total request timeout (I mean devices and backend components).

Reply:
So specifically it's to stop the api itself getting into a broken state due to all the waiting connection attempts when it's presumably already heavily overloaded.  There are other mechanisms around cancelling timed out requests and similar that handle stopping in-progress work when a request ends but this is a special case that we have less control over.

All that aside it's set up so that the values can be adjusted by any downstream users to fit their needs (which we already do for adjusting the pool size)

Reply:
Agreed, thanks for the explanation!

Reply:
I also want to start off conservative because right now we have no connection timeout at all, so I'd rather not set it too low by default and then find out the hard way